extensions/oplogPopulator/allocationStrategy/UniqueConnector.js

```diff
@@ -0,0 +1,44 @@
+const constants = require('../constants');
+const AllocationStrategy = require('./AllocationStrategy');
+
+/**
+ * @class UniqueConnector
+ *
+ * @classdesc UniqueConnector is an allocation
+ * strategy where each bucket is assigned to a unique
+ * connector.
+ */
+class UniqueConnector extends AllocationStrategy {
+
+    /**
+     * Get best connector to assign a bucket to.
+     * If no connector is available, null is returned.
+     * @param {Array<Connector>} connectors connectors
+     * @param {String} bucket bucket name
+     * @returns {Connector | null} connector
+     */
+    getConnector(connectors, bucket) { // eslint-disable-line no-unused-vars
+        if (!connectors.length) {
+            return null;
+        }
+        return connectors.find(conn => conn.config.pipeline?.includes(constants.wildCardForAllBuckets)) || null;
```

Comment:
Nit: Should we fail or crash when we don't find the connector ? This should never happen looking at the code. Also if we fall in that case and return `null` a new "wildcard" connector will be created but we're not certain no other connectors are laying around which will be a problem, maybe we should check that ?

Reply:
So we are supposed to delete any non-wildcard connector if this mode is used, and here we should detect any existing wildcard connector, which should cover all cases... If we do not detect the connector, we need to create it.